Core/include/Acts/EventData/TrackStateProxy.hpp

```diff
@@ -714,68 +749,36 @@ class TrackStateProxy {
   /// @warning The dynamic vector has a runtime overhead!
   /// @return The effective calibrated measurement vector
   template <bool RO = ReadOnly, typename = std::enable_if_t<!RO>>
-  auto effectiveCalibrated() {
-    // repackage the data pointer to a dynamic map type
-    // workaround for gcc8 bug:
-    // https://gcc.gnu.org/bugzilla/show_bug.cgi?id=86594
-    auto* self = this;
-    return visit_measurement(calibratedSize(), [&](auto N) {
-      constexpr int kMeasurementSize = decltype(N)::value;
-      return typename detail_lt::Types<M, ReadOnly>::DynamicCoefficientsMap{
-          self->template calibrated<kMeasurementSize>().data(),
-          kMeasurementSize};
-    });
+  EffectiveMeasurement effectiveCalibrated() {
+    assert(has<hashString("calibrated")>());
+    return m_traj->self().effectiveMeasurement(m_istate);
   }
 
   /// Const dynamic measurement vector with only the valid dimensions.
   /// @warning The dynamic matrix has a runtime overhead!
   /// @return The effective calibrated measurement vector
-  auto effectiveCalibrated() const {
-    // repackage the data pointer to a dynamic map type
-    // workaround for gcc8 bug:
-    // https://gcc.gnu.org/bugzilla/show_bug.cgi?id=86594
-    auto* self = this;
-    return visit_measurement(calibratedSize(), [&](auto N) {
-      constexpr int kMeasurementSize = decltype(N)::value;
-      return typename detail_lt::Types<M, true>::DynamicCoefficientsMap{
-          self->template calibrated<kMeasurementSize>().data(),
-          kMeasurementSize};
-    });
+  ConstEffectiveMeasurement effectiveCalibrated() const {
+    assert(has<hashString("calibrated")>());
+    return m_traj->self().effectiveMeasurement(m_istate);
```

Comment:
I'd suggest to keep the naming consistent, i.e. stick to `Calibrated`

Reply:
I noticed that in `MultiTrajectory` everything is called `measurement` so I sticked with that. Should we align these? Potentially in a different PR

Reply:
You're right. Never mind in that case.